<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Karissa Barthelson" />

<meta name="date" content="2021-10-22" />

<title>initialQC</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/yeti.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">2021-rosmap-ire</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="snakemake_pipeline.html">Pre-processing</a>
</li>
<li>
  <a href="first-analysis.html">QC</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/karissa-b/2021-rosmap-ire">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">initialQC</h1>
<h4 class="author">Karissa Barthelson</h4>
<h4 class="date">2021-10-22</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-10-25
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>2021-rosmap-ire/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211022code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211022)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211022code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211022)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomkarissab2021rosmapiretreed04ae021c6203085b980261ec26d5040457251cftargetblankd04ae02a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/karissa-b/2021-rosmap-ire/tree/d04ae021c6203085b980261ec26d5040457251cf" target="_blank">d04ae02</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomkarissab2021rosmapiretreed04ae021c6203085b980261ec26d5040457251cftargetblankd04ae02a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/karissa-b/2021-rosmap-ire/tree/d04ae021c6203085b980261ec26d5040457251cf" target="_blank">d04ae02</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rapp.history
    Ignored:    .Rproj.user/
    Ignored:    code/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/confidentialData/

Unstaged changes:
    Modified:   analysis/first-analysis.Rmd
    Modified:   analysis/snakemake_pipeline.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/first-analysis.Rmd</code>) and HTML (<code>docs/first-analysis.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/karissa-b/2021-rosmap-ire/blob/d04ae021c6203085b980261ec26d5040457251cf/analysis/first-analysis.Rmd" target="_blank">d04ae02</a>
</td>
<td>
Karissa Barthelson
</td>
<td>
2021-10-23
</td>
<td>
msore QC
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/karissa-b/2021-rosmap-ire/d04ae021c6203085b980261ec26d5040457251cf/docs/first-analysis.html" target="_blank">d04ae02</a>
</td>
<td>
Karissa Barthelson
</td>
<td>
2021-10-23
</td>
<td>
msore QC
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/karissa-b/2021-rosmap-ire/blob/189974b018f63d0e3cc03109d2fd18fc84ad3519/analysis/first-analysis.Rmd" target="_blank">189974b</a>
</td>
<td>
Karissa Barthelson
</td>
<td>
2021-10-23
</td>
<td>
more QC
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/karissa-b/2021-rosmap-ire/189974b018f63d0e3cc03109d2fd18fc84ad3519/docs/first-analysis.html" target="_blank">189974b</a>
</td>
<td>
Karissa Barthelson
</td>
<td>
2021-10-23
</td>
<td>
more QC
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/karissa-b/2021-rosmap-ire/blob/6acc95655ef8ba1e9dbaa394073da2b5d5e59439/analysis/first-analysis.Rmd" target="_blank">6acc956</a>
</td>
<td>
Karissa Barthelson
</td>
<td>
2021-10-23
</td>
<td>
added PCA
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/karissa-b/2021-rosmap-ire/6acc95655ef8ba1e9dbaa394073da2b5d5e59439/docs/first-analysis.html" target="_blank">6acc956</a>
</td>
<td>
Karissa Barthelson
</td>
<td>
2021-10-23
</td>
<td>
added PCA
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/karissa-b/2021-rosmap-ire/blob/eed2e7f249329ecc97fda181230b5eb3d83c408f/analysis/first-analysis.Rmd" target="_blank">eed2e7f</a>
</td>
<td>
Karissa Barthelson
</td>
<td>
2021-10-22
</td>
<td>
day 1
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/karissa-b/2021-rosmap-ire/eed2e7f249329ecc97fda181230b5eb3d83c408f/docs/first-analysis.html" target="_blank">eed2e7f</a>
</td>
<td>
Karissa Barthelson
</td>
<td>
2021-10-22
</td>
<td>
day 1
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/karissa-b/2021-rosmap-ire/blob/fa46de3b22ecbee19e57e1a04891585c5cb3277c/analysis/first-analysis.Rmd" target="_blank">fa46de3</a>
</td>
<td>
Karissa Barthelson
</td>
<td>
2021-10-22
</td>
<td>
day 1
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/karissa-b/2021-rosmap-ire/fa46de3b22ecbee19e57e1a04891585c5cb3277c/docs/first-analysis.html" target="_blank">fa46de3</a>
</td>
<td>
Karissa Barthelson
</td>
<td>
2021-10-22
</td>
<td>
day 1
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/karissa-b/2021-rosmap-ire/blob/dcd0ecfb39e94a48e83ea93e9f662e2a0f3b16c7/analysis/first-analysis.Rmd" target="_blank">dcd0ecf</a>
</td>
<td>
Karissa Barthelson
</td>
<td>
2021-10-22
</td>
<td>
anther commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/karissa-b/2021-rosmap-ire/dcd0ecfb39e94a48e83ea93e9f662e2a0f3b16c7/docs/first-analysis.html" target="_blank">dcd0ecf</a>
</td>
<td>
Karissa Barthelson
</td>
<td>
2021-10-22
</td>
<td>
anther commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/karissa-b/2021-rosmap-ire/c07e1e090511b146201ab2c03137b62f1c5be4fc/docs/first-analysis.html" target="_blank">c07e1e0</a>
</td>
<td>
Karissa Barthelson
</td>
<td>
2021-10-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/karissa-b/2021-rosmap-ire/da3a49300ba9766ab6dfaab116e6539d0ffee27b/docs/first-analysis.html" target="_blank">da3a493</a>
</td>
<td>
Karissa Barthelson
</td>
<td>
2021-10-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/karissa-b/2021-rosmap-ire/9583672f1c7939d16ce52807f3d4173a443476b7/docs/first-analysis.html" target="_blank">9583672</a>
</td>
<td>
Karissa Barthelson
</td>
<td>
2021-10-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/karissa-b/2021-rosmap-ire/bbc759ca02af291c9622521f757ef8a57139e9b5/docs/first-analysis.html" target="_blank">bbc759c</a>
</td>
<td>
Karissa Barthelson
</td>
<td>
2021-10-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/karissa-b/2021-rosmap-ire/a12648a4b0a681e81eda6e90c60cc364aaa8a6b8/docs/first-analysis.html" target="_blank">a12648a</a>
</td>
<td>
Karissa Barthelson
</td>
<td>
2021-10-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/karissa-b/2021-rosmap-ire/3a8906d3b7f9b4898eef645f4c46e484b3a34142/docs/first-analysis.html" target="_blank">3a8906d</a>
</td>
<td>
Karissa Barthelson
</td>
<td>
2021-10-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/karissa-b/2021-rosmap-ire/17422c83a44270325f4362e12c187ad37bbbe95b/docs/first-analysis.html" target="_blank">17422c8</a>
</td>
<td>
Karissa Barthelson
</td>
<td>
2021-10-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/karissa-b/2021-rosmap-ire/blob/8f3fd30a2ea3f4c0818902c8bccf3ee7ed98bcb2/analysis/first-analysis.Rmd" target="_blank">8f3fd30</a>
</td>
<td>
Karissa Barthelson
</td>
<td>
2021-10-22
</td>
<td>
Add my first analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(tidyverse)
library(magrittr)
library(pander)
library(AnnotationHub)
library(ggpubr)
library(ggfortify)
library(ggrepel)
library(ggeasy)
library(edgeR)
library(scales)
library(ngsReports)
library(pheatmap)
library(sva)</code></pre>
<pre class="r"><code>theme_set(theme_bw())
panderOptions(&quot;big.mark&quot;, &quot;,&quot;)
panderOptions(&quot;table.split.table&quot;, Inf)
panderOptions(&quot;table.style&quot;, &quot;rmarkdown&quot;)</code></pre>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<pre class="r"><code># Note that this chunk takes a while to run, and requires internet connection
ah &lt;- AnnotationHub() %&gt;%
    subset(species == &quot;Homo sapiens&quot;) %&gt;%
    subset(rdataclass == &quot;EnsDb&quot;)

ensDb &lt;- ah[[&quot;AH75011&quot;]] # for release 98, which is the genome version I aligned to
grTrans &lt;- transcripts(ensDb)
trLengths &lt;- exonsBy(ensDb, &quot;tx&quot;) %&gt;%
    width() %&gt;%
    vapply(sum, integer(1))
mcols(grTrans)$length &lt;- trLengths[names(grTrans)]
gcGene &lt;- grTrans %&gt;%
  mcols() %&gt;%
  as.data.frame() %&gt;%
  dplyr::select(gene_id, tx_id, gc_content, length) %&gt;%
  as_tibble() %&gt;%
  group_by(gene_id) %&gt;%
  summarise(
    gc_content = sum(gc_content*length) / sum(length),
    length = ceiling(median(length))
  )
grGenes &lt;- genes(ensDb)
mcols(grGenes) %&lt;&gt;%
  as.data.frame() %&gt;%
  left_join(gcGene) %&gt;%
  as.data.frame() %&gt;%
  DataFrame()</code></pre>
<p>In order to perform adequate QC, an <code>EnsDb</code> object was obtained for Ensembl release 98 using the <code>AnnotationHub</code> package. This provided the GC content and length for each of the 250,194 transcripts contained in that release.</p>
<pre class="r"><code>meta &lt;- 
  read_csv(&quot;data/confidentialData/ROSMAP-IN_assay_RNAseq_metadata.csv&quot;) %&gt;% 
  dplyr::select(specimenID, RIN, rnaBatch, libraryBatch, sequencingBatch) %&gt;% 
  left_join(read_csv(&quot;data/confidentialData/ROSMAP-IN_invitro_biospecimen_metadata.csv&quot;)) %&gt;% 
  left_join(read_csv(&quot;data/confidentialData/ROSMAP-IN_individual_metadata.csv&quot;)) %&gt;% 
  mutate(experimentor = str_extract(.$specimenID, 
                                    pattern = &quot;(AH|VL)&quot;), 
         temp = str_remove(.$specimenID, 
                           pattern = &quot;(AH|VL)&quot;) %&gt;% 
           str_remove(pattern = &quot;_BR[0-9]+&quot;), 
         sample = paste0(experimentor, &quot;.s&quot;, temp), 
         apoeGenotype = case_when(
           apoeGenotype == 22 ~ &quot;E2/E2&quot;, 
           apoeGenotype == 23 ~ &quot;E2/E3&quot;, 
           apoeGenotype == 33 ~ &quot;E3/E3&quot;, 
           apoeGenotype == 34 ~ &quot;E3/E4&quot;,
           apoeGenotype == 44 ~ &quot;E4/E4&quot;, 
         )
  ) %&gt;% 
  dplyr::select(sample, experimentor, everything())# make some new cols to match the names in the gene counts matrix


featureCounts &lt;- 
  read_delim(&quot;data/confidentialData/counts.out&quot;, delim = &quot;\t&quot;, skip = 1) %&gt;%
  set_names(basename(names(.))) %&gt;% 
  as.data.frame() %&gt;%
  dplyr::select(-c(Chr, Start, End, Length, Strand)) %&gt;% 
  as_tibble %&gt;% 
  column_to_rownames(&quot;Geneid&quot;)

# tidy up colnames
colnames(featureCounts) %&lt;&gt;% 
  str_remove(&quot;.Aligned.sortedByCoord.out.bam&quot;)</code></pre>
</div>
<div id="raw-data-qc" class="section level1">
<h1>raw data QC</h1>
<pre class="r"><code>fastqc_raw &lt;- list.files(
    path = &quot;data/confidentialData/fastqc_raw/&quot;,
    pattern = &quot;zip&quot;, 
    recursive = TRUE,
    full.names = TRUE
    ) %&gt;% 
    FastqcDataList()</code></pre>
<div id="gc-content" class="section level2">
<h2>GC content</h2>
<p>Strange peaks are observed with 100% GC content. After a quick google search, it appears these are sequeniccng artefacts.</p>
<pre class="r"><code>plotGcContent(
  x = fastqc_raw, 
  plotType = &quot;line&quot;,
  gcType = &quot;Transcriptome&quot;
) +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/first-analysis.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/karissa-b/2021-rosmap-ire/blob/fa46de3b22ecbee19e57e1a04891585c5cb3277c/docs/figure/first-analysis.Rmd/unnamed-chunk-2-1.png" target="_blank">fa46de3</a>
</td>
<td>
Karissa Barthelson
</td>
<td>
2021-10-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/karissa-b/2021-rosmap-ire/blob/403670c1344e85ee6a3a0efa0eada46e22bb524c/docs/figure/first-analysis.Rmd/unnamed-chunk-2-1.png" target="_blank">403670c</a>
</td>
<td>
Karissa Barthelson
</td>
<td>
2021-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="over-represented-sequences" class="section level2">
<h2>Over-represented sequences</h2>
<p>There doesn’t appear to be any over-represented sequences which are worrying. The top ones are mostly adaptors which are removed in the <code>snakemake</code> pipeline.</p>
<pre class="r"><code>getModule(fastqc_raw, &quot;Overrep&quot;) %&gt;% 
    group_by(Sequence, Possible_Source) %&gt;% 
    summarise(`Found In` = n(), `Highest Percentage` = max(Percentage)) %&gt;% 
    arrange(desc(`Highest Percentage`), desc(`Found In`)) %&gt;% 
    ungroup() %&gt;% 
    dplyr::slice(1:30) %&gt;%
    mutate(`Highest Percentage` = percent_format(0.01)(`Highest Percentage`/100)) %&gt;%
    pander(
        justify = &quot;llrr&quot;,
        caption = paste(
            &quot;*Top&quot;, nrow(.),&quot;Overrepresented sequences.&quot;,
            &quot;The number of samples they were found in is shown,&quot;,
            &quot;along with the percentage of the most &#39;contaminated&#39; sample.*&quot;
            )
    )</code></pre>
<table>
<caption><em>Top 22 Overrepresented sequences. The number of samples they were found in is shown, along with the percentage of the most ‘contaminated’ sample.</em></caption>
<colgroup>
<col width="51%" />
<col width="28%" />
<col width="6%" />
<col width="13%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Sequence</th>
<th align="left">Possible_Source</th>
<th align="right">Found In</th>
<th align="right">Highest Percentage</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">GGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGG</td>
<td align="left">No Hit</td>
<td align="right">37</td>
<td align="right">3.60%</td>
</tr>
<tr class="even">
<td align="left">CTTATACACATCTCCGAGCCCACGAGACGTAGAGGAATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAAAGGGGG</td>
<td align="left">TruSeq Adapter, Index 3 (96% over 28bp)</td>
<td align="right">3</td>
<td align="right">1.46%</td>
</tr>
<tr class="odd">
<td align="left">CTTATACACATCTCCGAGCCCACGAGACGGACTCCTATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAAAGGGGG</td>
<td align="left">RNA PCR Primer, Index 27 (96% over 30bp)</td>
<td align="right">1</td>
<td align="right">1.10%</td>
</tr>
<tr class="even">
<td align="left">CTTATACACATCTCCGAGCCCACGAGACGCTCATGAATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAAAGGGGG</td>
<td align="left">TruSeq Adapter, Index 8 (96% over 29bp)</td>
<td align="right">1</td>
<td align="right">1.01%</td>
</tr>
<tr class="odd">
<td align="left">ATACACATCTCCGAGCCCACGAGACGTAGAGGAATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAAAGGGGGGGG</td>
<td align="left">TruSeq Adapter, Index 3 (96% over 28bp)</td>
<td align="right">2</td>
<td align="right">0.87%</td>
</tr>
<tr class="even">
<td align="left">CTTATACACATCTCCGAGCCCACGAGACTAGGCATGATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAAAGGGGG</td>
<td align="left">TruSeq Adapter, Index 1 (96% over 28bp)</td>
<td align="right">1</td>
<td align="right">0.60%</td>
</tr>
<tr class="odd">
<td align="left">ATACACATCTCCGAGCCCACGAGACGCTCATGAATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAAAGGGGGGGG</td>
<td align="left">TruSeq Adapter, Index 8 (96% over 29bp)</td>
<td align="right">1</td>
<td align="right">0.59%</td>
</tr>
<tr class="even">
<td align="left">ATACACATCTCCGAGCCCACGAGACGGACTCCTATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAAAGGGGGGGG</td>
<td align="left">RNA PCR Primer, Index 27 (96% over 30bp)</td>
<td align="right">1</td>
<td align="right">0.58%</td>
</tr>
<tr class="odd">
<td align="left">ACACATCTCCGAGCCCACGAGACGTAGAGGAATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAAAGGGGGGGGGG</td>
<td align="left">RNA PCR Primer, Index 3 (96% over 28bp)</td>
<td align="right">1</td>
<td align="right">0.42%</td>
</tr>
<tr class="even">
<td align="left">ATACACATCTCCGAGCCCACGAGACTAGGCATGATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAAAGGGGGGGG</td>
<td align="left">TruSeq Adapter, Index 1 (96% over 28bp)</td>
<td align="right">1</td>
<td align="right">0.37%</td>
</tr>
<tr class="odd">
<td align="left">CTTATACACATCTCCGAGCCCACGAGACAAGAGGCAATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAAAGGGGG</td>
<td align="left">RNA PCR Primer, Index 33 (96% over 29bp)</td>
<td align="right">1</td>
<td align="right">0.29%</td>
</tr>
<tr class="even">
<td align="left">ACACATCTCCGAGCCCACGAGACGCTCATGAATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAAAGGGGGGGGGG</td>
<td align="left">RNA PCR Primer, Index 8 (96% over 29bp)</td>
<td align="right">1</td>
<td align="right">0.29%</td>
</tr>
<tr class="odd">
<td align="left">ACACATCTCCGAGCCCACGAGACGGACTCCTATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAAAGGGGGGGGGG</td>
<td align="left">RNA PCR Primer, Index 27 (96% over 30bp)</td>
<td align="right">1</td>
<td align="right">0.27%</td>
</tr>
<tr class="even">
<td align="left">CTTATACACATCTCCGAGCCCACGAGACCTCTCTACATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAAAGGGGG</td>
<td align="left">RNA PCR Primer, Index 39 (96% over 30bp)</td>
<td align="right">1</td>
<td align="right">0.26%</td>
</tr>
<tr class="odd">
<td align="left">CTTATACACATCTCCGAGCCCACGAGACCGAGGCTGATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAAAGGGGG</td>
<td align="left">TruSeq Adapter, Index 10 (96% over 28bp)</td>
<td align="right">2</td>
<td align="right">0.24%</td>
</tr>
<tr class="even">
<td align="left">ACACATCTCCGAGCCCACGAGACTAGGCATGATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAAAGGGGGGGGGG</td>
<td align="left">RNA PCR Primer, Index 1 (96% over 28bp)</td>
<td align="right">1</td>
<td align="right">0.20%</td>
</tr>
<tr class="odd">
<td align="left">ATACACATCTCCGAGCCCACGAGACAAGAGGCAATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAAAGGGGGGGG</td>
<td align="left">RNA PCR Primer, Index 33 (96% over 29bp)</td>
<td align="right">1</td>
<td align="right">0.18%</td>
</tr>
<tr class="even">
<td align="left">ATACACATCTCCGAGCCCACGAGACCTCTCTACATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAAAGGGGGGGG</td>
<td align="left">RNA PCR Primer, Index 39 (96% over 30bp)</td>
<td align="right">1</td>
<td align="right">0.15%</td>
</tr>
<tr class="odd">
<td align="left">ATACACATCTCCGAGCCCACGAGACCGAGGCTGATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAAAGGGGGGGG</td>
<td align="left">TruSeq Adapter, Index 10 (96% over 28bp)</td>
<td align="right">2</td>
<td align="right">0.15%</td>
</tr>
<tr class="even">
<td align="left">CTTATACACATCTCCGAGCCCACGAGACTCCTGAGCATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAAAGGGGG</td>
<td align="left">RNA PCR Primer, Index 26 (100% over 29bp)</td>
<td align="right">1</td>
<td align="right">0.14%</td>
</tr>
<tr class="odd">
<td align="left">CTTATACACATCTCCGAGCCCACGAGACCGTACTAGATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAAAGGGGG</td>
<td align="left">TruSeq Adapter, Index 11 (96% over 28bp)</td>
<td align="right">1</td>
<td align="right">0.11%</td>
</tr>
<tr class="even">
<td align="left">ACACATCTCCGAGCCCACGAGACCGAGGCTGATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAAAGGGGGGGGGG</td>
<td align="left">RNA PCR Primer, Index 10 (96% over 28bp)</td>
<td align="right">1</td>
<td align="right">0.10%</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="trimmed-data-qc" class="section level1">
<h1>Trimmed data QC</h1>
<pre class="r"><code>fastqc_trim &lt;- list.files(path = &quot;data/confidentialData/fastqc_trim&quot;, 
                          pattern = &quot;zip&quot;, 
                          full.names = TRUE) %&gt;%
  FastqcDataList()

trimStats &lt;- readTotals(fastqc_raw) %&gt;%
  dplyr::rename(Raw = Total_Sequences) %&gt;%
  left_join(readTotals(fastqc_trim), by = &quot;Filename&quot;) %&gt;%
  dplyr::rename(Trimmed = Total_Sequences) %&gt;%
  dplyr::filter(grepl(&quot;r1&quot;, Filename)) %&gt;%
  mutate(
    Discarded = 1 - Trimmed / Raw,
    Retained = Trimmed / Raw
  )</code></pre>
<p>The raw reads were processed with <a href="https://github.com/OpenGene/fastphttps://github.com/OpenGene/fastp">fastp</a>. Reads which contained an average quality score of &lt; 15 were omitted, as were reads which were shorter than 20nt after adaptor removal. I allowed fastp to detect adaptors automatically. After adapter trimming between <em>2.21%</em> and <em>8.67%</em> of reads were discarded. No over-represented sequences remained, and the %GC appears to have improved.</p>
<pre class="r"><code>ggarrange(
  plotGcContent(
    x = fastqc_raw, 
    plotType = &quot;line&quot;,
    gcType = &quot;Transcriptome&quot;
  ) +
    theme(legend.position = &quot;none&quot;) +
    ggtitle(&quot;Before trimming/filtering&quot;), 
  plotGcContent(
  x = fastqc_trim, 
  plotType = &quot;line&quot;,
  gcType = &quot;Transcriptome&quot;
) +
  theme(legend.position = &quot;none&quot;)+
  ggtitle(&quot;After trimming/filtering&quot;)
) </code></pre>
<p><img src="figure/first-analysis.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/karissa-b/2021-rosmap-ire/blob/eed2e7f249329ecc97fda181230b5eb3d83c408f/docs/figure/first-analysis.Rmd/unnamed-chunk-4-1.png" target="_blank">eed2e7f</a>
</td>
<td>
Karissa Barthelson
</td>
<td>
2021-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="align-qc" class="section level1">
<h1>Align QC</h1>
<p>Trimmed reads were aligned to the human genome (Ensembl relealease 98) using <code>STAR 2.7.0d</code> and summarised to each gene using <code>featureCounts</code>. These counts were to be used for all gene-level analysis.</p>
<pre class="r"><code>fastqc_align &lt;- list.files(path = &quot;data/confidentialData/fastqc_align&quot;, 
                          pattern = &quot;zip&quot;, 
                          full.names = TRUE) %&gt;%
  FastqcDataList()</code></pre>
<pre class="r"><code>plotGcContent(
  x = fastqc_align, 
  plotType = &quot;line&quot;,
  gcType = &quot;Transcriptome&quot;
) +
  theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/first-analysis.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/karissa-b/2021-rosmap-ire/blob/eed2e7f249329ecc97fda181230b5eb3d83c408f/docs/figure/first-analysis.Rmd/unnamed-chunk-6-1.png" target="_blank">eed2e7f</a>
</td>
<td>
Karissa Barthelson
</td>
<td>
2021-10-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/karissa-b/2021-rosmap-ire/blob/fa46de3b22ecbee19e57e1a04891585c5cb3277c/docs/figure/first-analysis.Rmd/unnamed-chunk-6-1.png" target="_blank">fa46de3</a>
</td>
<td>
Karissa Barthelson
</td>
<td>
2021-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="filtering-lowly-expressed-genes" class="section level1">
<h1>Filtering lowly expressed genes</h1>
<p>Genes which are lowly expressed are uninformative for DE analysis. Here, I will consider a gene to be lowly expressed if it contains a CPM of less than 2 in at least 33 samples (i.e. the number of AD samples). The effect of filtering is be shown in the density plots below.</p>
<pre class="r"><code>a &lt;- featureCounts %&gt;% 
  cpm(log = TRUE) %&gt;%
  as.data.frame() %&gt;% 
  pivot_longer(
    cols = everything(),
    names_to = &quot;sample&quot;,
    values_to = &quot;logCPM&quot;
  ) %&gt;%
  split(f = .$sample) %&gt;%
  lapply(function(x){
    d &lt;- density(x$logCPM)
    tibble(
      sample = unique(x$sample),
      x = d$x,
      y = d$y
    )
  }) %&gt;%
  bind_rows() %&gt;%
  left_join(meta) %&gt;% 
  ggplot(aes(x, y, colour = diagnosis, group = sample)) +
  geom_line() +
  labs(
    x = &quot;logCPM&quot;,
    y = &quot;Density&quot;,
    colour = &quot;diagnosis&quot;
  )+
  ggtitle(&quot;Before filtering&quot;)

b &lt;- featureCounts %&gt;% 
  .[rowSums(cpm(.) &gt;= 2) &gt;= 33,] %&gt;% 
  cpm(log = TRUE) %&gt;%
  as.data.frame() %&gt;% 
  pivot_longer(
    cols = everything(),
    names_to = &quot;sample&quot;,
    values_to = &quot;logCPM&quot;
  ) %&gt;%
  split(f = .$sample) %&gt;%
  lapply(function(x){
    d &lt;- density(x$logCPM)
    tibble(
      sample = unique(x$sample),
      x = d$x,
      y = d$y
    )
  }) %&gt;%
  bind_rows() %&gt;%
  left_join(meta) %&gt;% 
  ggplot(aes(x, y, colour = diagnosis, group = sample)) +
  geom_line() +
  labs(
    x = &quot;logCPM&quot;,
    y = &quot;Density&quot;,
    colour = &quot;Genotype&quot;
  )+
  ggtitle(&quot;After filtering&quot;) 
ggarrange(a, b, common.legend = TRUE)</code></pre>
<p><img src="figure/first-analysis.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/karissa-b/2021-rosmap-ire/blob/eed2e7f249329ecc97fda181230b5eb3d83c408f/docs/figure/first-analysis.Rmd/unnamed-chunk-7-1.png" target="_blank">eed2e7f</a>
</td>
<td>
Karissa Barthelson
</td>
<td>
2021-10-22
</td>
</tr>
<tr>
<td>
<a href="https://github.com/karissa-b/2021-rosmap-ire/blob/fa46de3b22ecbee19e57e1a04891585c5cb3277c/docs/figure/first-analysis.Rmd/unnamed-chunk-7-1.png" target="_blank">fa46de3</a>
</td>
<td>
Karissa Barthelson
</td>
<td>
2021-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>dge &lt;- featureCounts %&gt;% 
  as.matrix() %&gt;% 
  DGEList(
    samples = tibble(sample = colnames(.)) %&gt;%
      left_join(meta) %&gt;% 
      as_tibble(),
    genes = grGenes[rownames(.)] %&gt;%
      as.data.frame() %&gt;%
      dplyr::select(
        chromosome = seqnames, start, end, 
        gene_id, gene_name, gene_biotype, description, entrezid
      ) %&gt;% 
      left_join(gcGene) %&gt;% 
      as_tibble()
  ) %&gt;%
  calcNormFactors()</code></pre>
</div>
<div id="check-library-sizes" class="section level1">
<h1>Check library sizes</h1>
<p>Libary sizes vary considerably, which appears to be due to samples being sequenced across multiple runs. Library sizes range between <em>4,978,272</em> and <em>20,875,388</em>. Although concerning, this should be accounted for using <code>TMM normalisation</code>.</p>
<pre class="r"><code>dge$samples %&gt;% 
  ggplot(aes(x = specimenIdSource, y = lib.size, fill = sequencingBatch)) + 
  geom_col() +
  easy_rotate_x_labels(angle = -45) +
  facet_wrap(~diagnosis, scales = &quot;free_x&quot;) +
  scale_y_continuous(labels = comma) +
  labs(
    x = &quot;Sample&quot;, 
    y = &quot;RNA-seq library size&quot;, 
    fill = &quot;RNA-seq batch&quot;
  )</code></pre>
<p><img src="figure/first-analysis.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/karissa-b/2021-rosmap-ire/blob/eed2e7f249329ecc97fda181230b5eb3d83c408f/docs/figure/first-analysis.Rmd/unnamed-chunk-8-1.png" target="_blank">eed2e7f</a>
</td>
<td>
Karissa Barthelson
</td>
<td>
2021-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="pca" class="section level1">
<h1>PCA</h1>
<pre class="r"><code>dge &lt;- featureCounts %&gt;% 
  as.matrix() %&gt;% 
  .[rowSums(cpm(.) &gt;= 2) &gt;= 33,] %&gt;% 
  DGEList(
    samples = tibble(sample = colnames(.)) %&gt;%
      left_join(meta) %&gt;% 
      as_tibble(),
    genes = grGenes[rownames(.)] %&gt;%
      as.data.frame() %&gt;%
      dplyr::select(
        chromosome = seqnames, start, end, 
        gene_id, gene_name, gene_biotype, description, entrezid
      ) %&gt;% 
      left_join(gcGene) %&gt;% 
      as_tibble()
  ) %&gt;%
  calcNormFactors()</code></pre>
<p>I next performed a PCA on the logCPM values after filtering lowly expressed genes. I want to see how similar the samples look. Two distinct clusters are observed across PC2 (explaining ~19% of the total variation in the dataset).</p>
<pre class="r"><code>cpm(dge, log = TRUE) %&gt;%
  t() %&gt;%
  prcomp() %&gt;% 
  autoplot(data = tibble(sample = rownames(.$x)) %&gt;%
             left_join(dge$samples),
           colour = &quot;experimentor&quot;, 
           size = 4,
           shape = &quot;sex&quot;)</code></pre>
<p><img src="figure/first-analysis.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/karissa-b/2021-rosmap-ire/blob/6acc95655ef8ba1e9dbaa394073da2b5d5e59439/docs/figure/first-analysis.Rmd/unnamed-chunk-9-1.png" target="_blank">6acc956</a>
</td>
<td>
Karissa Barthelson
</td>
<td>
2021-10-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>I next want to see how similar the technical replicates are. In the plot below, only samples which were sequenced twice are labelled. Most technical replicates do not cluster together, meaning they are not overly similar.</p>
<pre class="r"><code>cpm(dge, log = TRUE) %&gt;%
  t() %&gt;%
  prcomp() %&gt;% 
  autoplot(data = tibble(sample = rownames(.$x)) %&gt;%
             left_join(dge$samples),
             size = 2,
             shape = &quot;sex&quot;) +
    geom_label_repel(aes(label = specimenIdSource, 
                         colour = specimenIdSource), 
                     max.overlaps = 15,
                     data = . %&gt;% 
                       dplyr::filter(specimenIdSource %in% dge$samples$specimenIdSource[dge$samples %&gt;% dplyr::select(&quot;specimenIdSource&quot;) %&gt;% duplicated()])
                     )</code></pre>
<p><img src="figure/first-analysis.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/karissa-b/2021-rosmap-ire/blob/6acc95655ef8ba1e9dbaa394073da2b5d5e59439/docs/figure/first-analysis.Rmd/unnamed-chunk-10-1.png" target="_blank">6acc956</a>
</td>
<td>
Karissa Barthelson
</td>
<td>
2021-10-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="why-do-the-technical-replicates-look-so-different" class="section level1">
<h1>Why do the technical replicates look so different?</h1>
<p>One would expect that technical replicates would look mostly similar since they were derived from the same sample of iPSC neurons. But the PCA plot above shows that this is not the case. From inspection of the metadata about the samples, looks like the RNA integrity number (RIN) is fairly low in one of the library preparation batches.</p>
<pre class="r"><code>techRep_brID &lt;- dge$samples$specimenIdSource[dge$samples %&gt;% dplyr::select(&quot;specimenIdSource&quot;) %&gt;% duplicated()]

techRep_sampleID &lt;- dge$samples %&gt;% 
  dplyr::filter(specimenIdSource %in% techRep_brID) %&gt;% 
  .$sample
 
dge$samples %&gt;%  
  dplyr::filter(specimenIdSource %in% techRep_brID) %&gt;% 
  ggplot(aes(x = sample, y = lib.size/1e6, fill = RIN)) +
  geom_col() +
  facet_wrap(~libraryBatch, scales = &quot;free_x&quot;) +
  scale_fill_viridis_c() +
  labs(y = &quot;Library size (millions)&quot;) +
  easy_rotate_x_labels(angle = -45) +
  ggtitle(&quot;Library sizes of samples which had technical replicates,\ncoloured by RNA integrity&quot;, 
          subtitle = &quot;Samples are grouped by library batch&quot;)</code></pre>
<p><img src="figure/first-analysis.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/karissa-b/2021-rosmap-ire/blob/189974b018f63d0e3cc03109d2fd18fc84ad3519/docs/figure/first-analysis.Rmd/unnamed-chunk-11-1.png" target="_blank">189974b</a>
</td>
<td>
Karissa Barthelson
</td>
<td>
2021-10-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Inspection of the two replicates for each iPSC neuron samples shows that most of the time (except for BR14, which had fairly similar RINs), there is onw “high quality” library, and one with lesser quality.</p>
<pre class="r"><code>dge$samples %&gt;%  
  dplyr::filter(specimenIdSource %in% techRep_brID) %&gt;% 
  ggplot(aes(x = sample, y = lib.size/1e6, fill = RIN)) +
  geom_col() +
  facet_wrap(~specimenIdSource, scales = &quot;free_x&quot;) +
  scale_fill_viridis_c() +
  labs(y = &quot;Library size (millions)&quot;) +
  easy_rotate_x_labels(angle = -45)</code></pre>
<p><img src="figure/first-analysis.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/karissa-b/2021-rosmap-ire/blob/189974b018f63d0e3cc03109d2fd18fc84ad3519/docs/figure/first-analysis.Rmd/unnamed-chunk-12-1.png" target="_blank">189974b</a>
</td>
<td>
Karissa Barthelson
</td>
<td>
2021-10-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>samps2keep &lt;- dge$samples %&gt;% 
  group_by(specimenIdSource) %&gt;% 
  dplyr::slice(which.max(RIN)) %&gt;% 
  .$sample
dge &lt;- dge[,samps2keep]</code></pre>
<p>Therefore, for this analysis, I will omit the technical replicate which had the lower RIN for each iPSC sample. This leaves 49 samples out of the original 66. This is a different strategy to <a href="https://doi.org/10.1016/j.neuron.2021.08.003">Lagomarsino et al. 2021</a>, who averaged out their replicates. I dont think this is the best way to deal with this, as the RNA was degraded in one of the technical replicates. This may influence how the reads are generated during the sequencing, and avergaing it out across a better quality sample may bring about false results.</p>
</div>
<div id="pca-after-omitting-samples" class="section level1">
<h1>PCA after omitting samples</h1>
<p>The PCA was repeated after omitting the problematic RNA-seq libraries. Samples still seperate by library batch. So I will need to correct for this.</p>
<pre class="r"><code>cpm(dge, log = TRUE) %&gt;%
  t() %&gt;%
  prcomp() %&gt;% 
  autoplot(data = tibble(sample = rownames(.$x)) %&gt;%
             left_join(dge$samples),
           colour = &quot;diagnosis&quot;, 
           size = 4) +
  scale_colour_viridis_d(end = 0.8) +
    ggtitle(&quot;Diagnosis&quot;)</code></pre>
<p><img src="figure/first-analysis.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/karissa-b/2021-rosmap-ire/blob/d04ae021c6203085b980261ec26d5040457251cf/docs/figure/first-analysis.Rmd/unnamed-chunk-14-1.png" target="_blank">d04ae02</a>
</td>
<td>
Karissa Barthelson
</td>
<td>
2021-10-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>cpm(dge, log = TRUE) %&gt;%
  t() %&gt;%
  prcomp() %&gt;% 
  autoplot(data = tibble(sample = rownames(.$x)) %&gt;%
             left_join(dge$samples),
           colour = &quot;sex&quot;, 
           size = 4) +
  scale_colour_viridis_d(end = 0.8) +
  ggtitle(&quot;Sex&quot;)</code></pre>
<p><img src="figure/first-analysis.Rmd/unnamed-chunk-14-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>cpm(dge, log = TRUE) %&gt;%
  t() %&gt;%
  prcomp() %&gt;% 
  autoplot(data = tibble(sample = rownames(.$x)) %&gt;%
             left_join(dge$samples),
           colour = &quot;libraryBatch&quot;, 
           size = 4) +
  scale_colour_viridis_d(end = 0.9) +
  ggtitle(&quot;RNA-seq library batch&quot;)</code></pre>
<p><img src="figure/first-analysis.Rmd/unnamed-chunk-14-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>cpm(dge, log = TRUE) %&gt;%
  t() %&gt;%
  prcomp() %&gt;% 
  autoplot(data = tibble(sample = rownames(.$x)) %&gt;%
             left_join(dge$samples),
           colour = &quot;RIN&quot;, 
           size = 4) +
  scale_color_viridis_c() +
  ggtitle(&quot;RNA Integrity Number (RIN)&quot;)</code></pre>
<p><img src="figure/first-analysis.Rmd/unnamed-chunk-14-4.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="combat-to-remove-batch-effects" class="section level1">
<h1>ComBat to remove batch effects</h1>
<pre class="r"><code># define the batch effect to remove. 
batch &lt;- dge$samples$libraryBatch

adjustedCounts &lt;- ComBat_seq(counts = dge$counts, 
                             batch = batch, 
                             covar_mod  = dge$samples %&gt;% 
                               mutate(Braak = as.factor(Braak), 
                                      CERAD = as.factor(CERAD)) %&gt;% 
                               dplyr::select(diagnosis, sex, apoeGenotype, 
                                             yearsEducation, CERAD, Braak) %&gt;%
                               as.data.frame()
)</code></pre>
<pre><code>Found 5 batches
Using null model in ComBat-seq.
Adjusting for 15 covariate(s) or covariate level(s)
Estimating dispersions
Fitting the GLM model
Shrinkage off - using GLM estimates for parameters
Adjusting the data</code></pre>
<pre class="r"><code># repeat the PCA
adjustedCounts %&gt;% 
  cpm(log = TRUE) %&gt;%
  t() %&gt;%
  prcomp() %&gt;% 
  autoplot(data = tibble(sample = rownames(.$x)) %&gt;%
             left_join(dge$samples),
           colour = &quot;diagnosis&quot;, 
           shape = &quot;sex&quot;,
           size = 4) +
  scale_color_viridis_d() +
  theme(aspect.ratio = 1)</code></pre>
<p><img src="figure/first-analysis.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/karissa-b/2021-rosmap-ire/blob/d04ae021c6203085b980261ec26d5040457251cf/docs/figure/first-analysis.Rmd/unnamed-chunk-15-1.png" target="_blank">d04ae02</a>
</td>
<td>
Karissa Barthelson
</td>
<td>
2021-10-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.2 (2020-06-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Mojave 10.14.3

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] ensembldb_2.12.1        AnnotationFilter_1.12.0 GenomicFeatures_1.40.1 
 [4] AnnotationDbi_1.50.3    Biobase_2.48.0          GenomicRanges_1.40.0   
 [7] GenomeInfoDb_1.24.2     IRanges_2.22.2          S4Vectors_0.26.1       
[10] sva_3.36.0              BiocParallel_1.22.0     genefilter_1.70.0      
[13] mgcv_1.8-36             nlme_3.1-152            pheatmap_1.0.12        
[16] ngsReports_1.4.2        scales_1.1.1            edgeR_3.30.3           
[19] limma_3.44.3            ggeasy_0.1.3            ggrepel_0.9.1          
[22] ggfortify_0.4.12        ggpubr_0.4.0            AnnotationHub_2.20.2   
[25] BiocFileCache_1.12.1    dbplyr_2.1.1            BiocGenerics_0.34.0    
[28] pander_0.6.4            magrittr_2.0.1          forcats_0.5.1          
[31] stringr_1.4.0           dplyr_1.0.7             purrr_0.3.4            
[34] readr_1.4.0             tidyr_1.1.3             tibble_3.1.2           
[37] ggplot2_3.3.5           tidyverse_1.3.1         workflowr_1.6.2        

loaded via a namespace (and not attached):
  [1] readxl_1.3.1                  backports_1.2.1              
  [3] plyr_1.8.6                    lazyeval_0.2.2               
  [5] splines_4.0.2                 digest_0.6.27                
  [7] htmltools_0.5.1.1             fansi_0.5.0                  
  [9] memoise_2.0.0                 cluster_2.1.2                
 [11] openxlsx_4.2.4                annotate_1.66.0              
 [13] Biostrings_2.56.0             modelr_0.1.8                 
 [15] matrixStats_0.59.0            askpass_1.1                  
 [17] prettyunits_1.1.1             jpeg_0.1-8.1                 
 [19] colorspace_2.0-2              blob_1.2.1                   
 [21] rvest_1.0.0                   rappdirs_0.3.3               
 [23] haven_2.4.1                   xfun_0.24                    
 [25] crayon_1.4.1                  RCurl_1.98-1.3               
 [27] jsonlite_1.7.2                survival_3.2-11              
 [29] zoo_1.8-9                     glue_1.4.2                   
 [31] gtable_0.3.0                  zlibbioc_1.34.0              
 [33] XVector_0.28.0                DelayedArray_0.14.1          
 [35] car_3.0-11                    abind_1.4-5                  
 [37] DBI_1.1.1                     rstatix_0.7.0                
 [39] Rcpp_1.0.7                    progress_1.2.2               
 [41] viridisLite_0.4.0             xtable_1.8-4                 
 [43] flashClust_1.01-2             foreign_0.8-81               
 [45] bit_4.0.4                     DT_0.18                      
 [47] truncnorm_1.0-8               htmlwidgets_1.5.3            
 [49] httr_1.4.2                    RColorBrewer_1.1-2           
 [51] ellipsis_0.3.2                farver_2.1.0                 
 [53] XML_3.99-0.6                  pkgconfig_2.0.3              
 [55] sass_0.4.0                    locfit_1.5-9.4               
 [57] utf8_1.2.1                    labeling_0.4.2               
 [59] tidyselect_1.1.1              rlang_0.4.11                 
 [61] reshape2_1.4.4                later_1.2.0                  
 [63] munsell_0.5.0                 BiocVersion_3.11.1           
 [65] cellranger_1.1.0              tools_4.0.2                  
 [67] cachem_1.0.5                  cli_3.0.0                    
 [69] generics_0.1.0                RSQLite_2.2.7                
 [71] broom_0.7.8                   evaluate_0.14                
 [73] fastmap_1.1.0                 ggdendro_0.1.22              
 [75] yaml_2.2.1                    knitr_1.33                   
 [77] bit64_4.0.5                   fs_1.5.0                     
 [79] zip_2.2.0                     whisker_0.4                  
 [81] mime_0.11                     leaps_3.1                    
 [83] xml2_1.3.2                    biomaRt_2.44.4               
 [85] compiler_4.0.2                rstudioapi_0.13              
 [87] png_0.1-7                     plotly_4.9.4.1               
 [89] curl_4.3.2                    interactiveDisplayBase_1.26.3
 [91] ggsignif_0.6.2                reprex_2.0.0                 
 [93] bslib_0.2.5.1                 stringi_1.6.2                
 [95] highr_0.9                     lattice_0.20-44              
 [97] ProtGenerics_1.20.0           Matrix_1.3-4                 
 [99] vctrs_0.3.8                   pillar_1.6.1                 
[101] lifecycle_1.0.0               BiocManager_1.30.16          
[103] jquerylib_0.1.4               cowplot_1.1.1                
[105] data.table_1.14.0             bitops_1.0-7                 
[107] rtracklayer_1.48.0            httpuv_1.6.1                 
[109] latticeExtra_0.6-29           hwriter_1.3.2                
[111] R6_2.5.0                      ShortRead_1.46.0             
[113] promises_1.2.0.1              gridExtra_2.3                
[115] rio_0.5.27                    MASS_7.3-54                  
[117] assertthat_0.2.1              SummarizedExperiment_1.18.2  
[119] openssl_1.4.4                 rprojroot_2.0.2              
[121] withr_2.4.2                   GenomicAlignments_1.24.0     
[123] Rsamtools_2.4.0               GenomeInfoDbData_1.2.3       
[125] hms_1.1.0                     grid_4.0.2                   
[127] rmarkdown_2.9                 carData_3.0-4                
[129] git2r_0.28.0                  scatterplot3d_0.3-41         
[131] shiny_1.6.0                   lubridate_1.7.10             
[133] FactoMineR_2.4               </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
