<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Karissa Barthelson" />

<meta name="date" content="2021-10-25" />

<title>removeBatchEffects</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/yeti.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">2021-rosmap-ire</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="snakemake_pipeline.html">Pre-processing</a>
</li>
<li>
  <a href="first-analysis.html">QC</a>
</li>
<li>
  <a href="removeBatchEffects.html">Remove batch effects</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/karissa-b/2021-rosmap-ire">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">removeBatchEffects</h1>
<h4 class="author">Karissa Barthelson</h4>
<h4 class="date">2021-10-25</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-10-27
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>2021-rosmap-ire/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211022code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211022)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211022code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211022)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomkarissab2021rosmapiretree0e4608868d091856a8715286986e3d3f859947fatargetblank0e46088a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/karissa-b/2021-rosmap-ire/tree/0e4608868d091856a8715286986e3d3f859947fa" target="_blank">0e46088</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomkarissab2021rosmapiretree0e4608868d091856a8715286986e3d3f859947fatargetblank0e46088a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/karissa-b/2021-rosmap-ire/tree/0e4608868d091856a8715286986e3d3f859947fa" target="_blank">0e46088</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rapp.history
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    code/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/confidentialData/

Unstaged changes:
    Modified:   analysis/_site.yml
    Modified:   analysis/removeBatchEffects.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/removeBatchEffects.Rmd</code>) and HTML (<code>docs/removeBatchEffects.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/karissa-b/2021-rosmap-ire/blob/0e4608868d091856a8715286986e3d3f859947fa/analysis/removeBatchEffects.Rmd" target="_blank">0e46088</a>
</td>
<td>
Karissa Barthelson
</td>
<td>
2021-10-25
</td>
<td>
initial QC done?
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/karissa-b/2021-rosmap-ire/0e4608868d091856a8715286986e3d3f859947fa/docs/removeBatchEffects.html" target="_blank">0e46088</a>
</td>
<td>
Karissa Barthelson
</td>
<td>
2021-10-25
</td>
<td>
initial QC done?
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="exploration-of-removal-of-batch-effects" class="section level2">
<h2>Exploration of removal of batch effects</h2>
<pre class="r"><code>library(tidyverse)
library(magrittr)
library(edgeR)
library(sva)
library(pander)
library(scales)
library(pheatmap)
library(ggpubr)
library(ggfortify)
library(ggrepel)
library(ggeasy)

theme_set(theme_bw())
panderOptions(&quot;big.mark&quot;, &quot;,&quot;)
panderOptions(&quot;table.split.table&quot;, Inf)
panderOptions(&quot;table.style&quot;, &quot;rmarkdown&quot;)</code></pre>
<p>In this analysis, I will explore the effects of removing the batch effects present during the generation of this dataset. In the <a href="first-analysis.html">initial QC doc</a>, I noted that library preparation/RNA-seq batch still appears to drive some of the variation in this dataset. <a href="https://doi.org/10.1016/j.neuron.2021.08.003">Lagomarsino et al. 2021</a> noted that they used ComBat from the <code>sva</code> package to remove batch effects. So I will try this too.</p>
<pre class="r"><code>dge &lt;- readRDS(&quot;data/confidentialData/dge.rds&quot;)

# edit the covariates to be factors rather than character/integers
dge$samples %&lt;&gt;% 
  dplyr::select(1:11, diagnosis, apoeGenotype, pmi, sex, yearsEducation, CERAD, Braak) %&gt;% 
  mutate(diagnosis = case_when(
    diagnosis == &quot;no cognitive impairment&quot; ~ &quot;NCI&quot;, 
    diagnosis == &quot;Alzheimer Disease&quot; ~ &quot;AD&quot; 
  ) %&gt;% 
    factor(levels = c(&quot;NCI&quot;, &quot;AD&quot;)), 
  
  sex = as.factor(sex), 
  
  apoeGenotype = factor(apoeGenotype, levels = c(&quot;E3/E3&quot;, &quot;E2/E3&quot;, &quot;E3/E4&quot; ,&quot;E4/E4&quot;, &quot;E2/E2&quot;)), 
  CERAD = as.factor(CERAD), 
  Braak = as.factor(Braak)
    ) </code></pre>
<p>I imported the <code>dge</code> object generated in the <a href="first-analysis.html">initial QC doc</a>. This <code>dge</code> object contains the expression values per sample after:</p>
<ul>
<li>filtering lowly expressed genes (i.e. genes which have a logCPM &lt; 2)</li>
<li>Omitting one of the technical replicates of each iPSC sample which had the lower RIN</li>
</ul>
</div>
<div id="combat-to-remove-batch-effects" class="section level1">
<h1>ComBat to remove batch effects</h1>
<p>The ComBat-Seq function of the sva package is designed to remove batch effects from RNA-seq data. It uses a negative binomial regression to model the count matrix, and estimate parameters representing the batch effects. Then it provides adjusted data by mapping the original data to an expected distribution if there were no batch effects. The adjusted data preserve the integer nature of count matrix. Like ComBat, it requires known a batch variable, in this case, the library preparation batch.</p>
<p>ComBat-Seq allows specification of biological covariates whose signals will be preserved in the adjusted data. I will compare the results in a principal component analysis with and without including covariates (i.e. diagnosis, sex). Running ComBat-seq does appear to remove most of the variation due to library batch.</p>
<pre class="r"><code># make a list of three possible options in removing batch effects
adjustedCounts &lt;- list(nocovar = ComBat_seq(dge$counts, 
                                                     batch=dge$samples$libraryBatch, 
                                                     group=NULL), 
                       covar_diagnosisOnly = ComBat_seq(dge$counts, 
                                                        batch=dge$samples$libraryBatch, 
                                                        group=dge$samples$diagnosis
                       ), allCovar = ComBat_seq(dge$counts, 
                                                batch = dge$samples$libraryBatch, 
                                                covar_mod = dge$samples %&gt;% 
                                                  dplyr::select(diagnosis, sex, apoeGenotype, Braak, CERAD) %&gt;% 
                                                  as.data.frame()
                       )
)</code></pre>
<pre><code>Found 5 batches
Using null model in ComBat-seq.
Adjusting for 0 covariate(s) or covariate level(s)
Estimating dispersions
Fitting the GLM model
Shrinkage off - using GLM estimates for parameters
Adjusting the data
Found 5 batches
Using full model in ComBat-seq.
Adjusting for 1 covariate(s) or covariate level(s)
Estimating dispersions
Fitting the GLM model
Shrinkage off - using GLM estimates for parameters
Adjusting the data
Found 5 batches
Using null model in ComBat-seq.
Adjusting for 14 covariate(s) or covariate level(s)
Estimating dispersions
Fitting the GLM model
Shrinkage off - using GLM estimates for parameters
Adjusting the data</code></pre>
<pre class="r"><code>PCAplots &lt;- adjustedCounts %&gt;% 
  lapply(function(x) {
    cpm(x, log = TRUE) %&gt;%
      t() %&gt;%
      prcomp() %&gt;% 
      autoplot(data = tibble(sample = rownames(.$x)) %&gt;%
                 left_join(dge$samples),
               colour = &quot;libraryBatch&quot;, 
               shape = &quot;diagnosis&quot;,
               size = 4) +
      # scale_colour_viridis_d(end = 0.9) +
      theme(aspect.ratio = 1) +
      ggtitle(&quot;libraryBatch&quot;)
  }
  )
ggarrange(
  PCAplots$nocovar +
    ggtitle(&quot;No covariates&quot;), 
  
  PCAplots$covar_diagnosisOnly +
  ggtitle(&quot;Diagnosis only as a covariate&quot;), 
  
  PCAplots$allCovar +
  ggtitle(&quot;Covariates specified as\ndiagnosis, sex, apoeGenotype, Braak, CERAD&quot;), 
  common.legend = TRUE
)</code></pre>
<p><img src="figure/removeBatchEffects.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>After some reading about ComBat, I noted that “not including covariates may introduce bias or lead to the removal of biological signal”. Therefore, if I were to use ComBat to remove batch effects, I should use the method where I retain all co-variates.</p>
<p>However, I found this paper by <a href="https://doi.org/10.1093/biostatistics/kxv027">Nygaard et al. 2016</a>, who showed that removing batch effects <em>before</em> performing an analysis between groups (e.g. differential gene expression analysis) is actually not the most ideal method. Especially when the groups of interest are unbalanced within batches. Unfortunately, this is the case in this experiment (see graph below).</p>
<pre class="r"><code>dge$samples %&gt;% 
  dplyr::select(sample, libraryBatch, diagnosis, sex) %&gt;% 
  group_by(libraryBatch, diagnosis, sex) %&gt;% 
  summarise(n()) %&gt;% 
  ggplot(aes(x = libraryBatch, y = `n()`)) +
  geom_col(aes(fill = diagnosis), 
           colour = &quot;black&quot;) +
  facet_wrap(~sex) +
  labs(y = &quot;Number of samples&quot;) + 
  scale_fill_viridis_d()</code></pre>
<p><img src="figure/removeBatchEffects.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="conclusion" class="section level1">
<h1>Conclusion</h1>
<p>Because we have unbalanced groups of interest within batches (i.e. library prep batch), removing batch effects via ComBat should not be performed. Instead, I will include the effect of library batch in the model for differential gene expression analysis following the instructions in <code>edgeRUsersGuide</code> (section 3.4.3 Batch effects).</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.2 (2020-06-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Mojave 10.14.3

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] ggeasy_0.1.3        ggrepel_0.9.1       ggfortify_0.4.12   
 [4] ggpubr_0.4.0        pheatmap_1.0.12     scales_1.1.1       
 [7] pander_0.6.4        sva_3.36.0          BiocParallel_1.22.0
[10] genefilter_1.70.0   mgcv_1.8-36         nlme_3.1-152       
[13] edgeR_3.30.3        limma_3.44.3        magrittr_2.0.1     
[16] forcats_0.5.1       stringr_1.4.0       dplyr_1.0.7        
[19] purrr_0.3.4         readr_1.4.0         tidyr_1.1.3        
[22] tibble_3.1.2        ggplot2_3.3.5       tidyverse_1.3.1    

loaded via a namespace (and not attached):
  [1] colorspace_2.0-2     ggsignif_0.6.2       ellipsis_0.3.2      
  [4] rio_0.5.27           rprojroot_2.0.2      fs_1.5.0            
  [7] rstudioapi_0.13      farver_2.1.0         bit64_4.0.5         
 [10] AnnotationDbi_1.50.3 fansi_0.5.0          lubridate_1.7.10    
 [13] xml2_1.3.2           splines_4.0.2        cachem_1.0.5        
 [16] knitr_1.33           jsonlite_1.7.2       workflowr_1.6.2     
 [19] broom_0.7.8          annotate_1.66.0      dbplyr_2.1.1        
 [22] compiler_4.0.2       httr_1.4.2           backports_1.2.1     
 [25] assertthat_0.2.1     Matrix_1.3-4         fastmap_1.1.0       
 [28] cli_3.0.0            later_1.2.0          htmltools_0.5.1.1   
 [31] tools_4.0.2          gtable_0.3.0         glue_1.4.2          
 [34] Rcpp_1.0.7           carData_3.0-4        Biobase_2.48.0      
 [37] cellranger_1.1.0     jquerylib_0.1.4      vctrs_0.3.8         
 [40] xfun_0.24            openxlsx_4.2.4       rvest_1.0.0         
 [43] lifecycle_1.0.0      rstatix_0.7.0        XML_3.99-0.6        
 [46] hms_1.1.0            promises_1.2.0.1     parallel_4.0.2      
 [49] RColorBrewer_1.1-2   yaml_2.2.1           curl_4.3.2          
 [52] memoise_2.0.0        gridExtra_2.3        sass_0.4.0          
 [55] stringi_1.6.2        RSQLite_2.2.7        highr_0.9           
 [58] S4Vectors_0.26.1     BiocGenerics_0.34.0  zip_2.2.0           
 [61] rlang_0.4.11         pkgconfig_2.0.3      bitops_1.0-7        
 [64] matrixStats_0.59.0   evaluate_0.14        lattice_0.20-44     
 [67] labeling_0.4.2       cowplot_1.1.1        bit_4.0.4           
 [70] tidyselect_1.1.1     R6_2.5.0             IRanges_2.22.2      
 [73] generics_0.1.0       DBI_1.1.1            pillar_1.6.1        
 [76] haven_2.4.1          whisker_0.4          foreign_0.8-81      
 [79] withr_2.4.2          survival_3.2-11      abind_1.4-5         
 [82] RCurl_1.98-1.3       modelr_0.1.8         crayon_1.4.1        
 [85] car_3.0-11           utf8_1.2.1           rmarkdown_2.9       
 [88] locfit_1.5-9.4       grid_4.0.2           readxl_1.3.1        
 [91] data.table_1.14.0    blob_1.2.1           git2r_0.28.0        
 [94] reprex_2.0.0         digest_0.6.27        xtable_1.8-4        
 [97] httpuv_1.6.1         stats4_4.0.2         munsell_0.5.0       
[100] viridisLite_0.4.0    bslib_0.2.5.1       </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
