<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Karissa Barthelson" />

<meta name="date" content="2021-10-23" />

<title>snakemake_pipeline</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/yeti.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">2021-human-AD-iNs-IRE/</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    iPSC-neurons
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="snakemake_pipeline.html">Pre-processing</a>
    </li>
    <li>
      <a href="first-analysis.html">QC</a>
    </li>
    <li>
      <a href="removeBatchEffects.html">Remove batch effects</a>
    </li>
    <li>
      <a href="DGE.html">DGE analysis exploration of methods</a>
    </li>
    <li>
      <a href="ire_test.html">IRE enrichment analysis</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/karissa-b/2021-human-AD-iNs-IRE/">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">snakemake_pipeline</h1>
<h4 class="author">Karissa Barthelson</h4>
<h4 class="date">2021-10-23</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-10-29
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>2021-rosmap-ire/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211022code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211022)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211022code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211022)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomkarissab2021humanADiNsIREtreee988a78fb0670075d7761a77f372a0151e4157bbtargetblanke988a78a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/karissa-b/2021-human-AD-iNs-IRE/tree/e988a78fb0670075d7761a77f372a0151e4157bb" target="_blank">e988a78</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomkarissab2021humanADiNsIREtreee988a78fb0670075d7761a77f372a0151e4157bbtargetblanke988a78a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/karissa-b/2021-human-AD-iNs-IRE/tree/e988a78fb0670075d7761a77f372a0151e4157bb" target="_blank">e988a78</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rapp.history
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    code/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/confidentialData/

Staged changes:
    Modified:   analysis/_site.yml

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/snakemake_pipeline.Rmd</code>) and HTML (<code>docs/snakemake_pipeline.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/karissa-b/2021-human-AD-iNs-IRE/e7843ab0be6bd4674eda55f65e57e5d697e85526/docs/snakemake_pipeline.html" target="_blank">e7843ab</a>
</td>
<td>
Karissa Barthelson
</td>
<td>
2021-10-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/karissa-b/2021-human-AD-iNs-IRE/0306966d9cae43afe60e0c64349ddc0a5f7c6d89/docs/snakemake_pipeline.html" target="_blank">0306966</a>
</td>
<td>
Karissa Barthelson
</td>
<td>
2021-10-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/karissa-b/2021-human-AD-iNs-IRE/9d7aed542969d005f5e9a73fa46956190937e02c/docs/snakemake_pipeline.html" target="_blank">9d7aed5</a>
</td>
<td>
Karissa Barthelson
</td>
<td>
2021-10-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/karissa-b/2021-human-AD-iNs-IRE/blob/d915949dbd7a42f6eb1104f2c6129b32a9b658a2/analysis/snakemake_pipeline.Rmd" target="_blank">d915949</a>
</td>
<td>
Karissa Barthelson
</td>
<td>
2021-10-25
</td>
<td>
more an
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/karissa-b/2021-human-AD-iNs-IRE/d915949dbd7a42f6eb1104f2c6129b32a9b658a2/docs/snakemake_pipeline.html" target="_blank">d915949</a>
</td>
<td>
Karissa Barthelson
</td>
<td>
2021-10-25
</td>
<td>
more an
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/karissa-b/2021-human-AD-iNs-IRE/blob/d04ae021c6203085b980261ec26d5040457251cf/analysis/snakemake_pipeline.Rmd" target="_blank">d04ae02</a>
</td>
<td>
Karissa Barthelson
</td>
<td>
2021-10-23
</td>
<td>
msore QC
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/karissa-b/2021-human-AD-iNs-IRE/d04ae021c6203085b980261ec26d5040457251cf/docs/snakemake_pipeline.html" target="_blank">d04ae02</a>
</td>
<td>
Karissa Barthelson
</td>
<td>
2021-10-23
</td>
<td>
msore QC
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>This document shows the <code>Snakemake</code> pipeline used to pre-process the raw fastq files to a gene counts matrix. I still need to add the default.yaml file, which contains all the versions etc.</p>
<pre class="r"><code>cat(readLines(&quot;Snakefile&quot;), sep = &quot;\n&quot;)</code></pre>
<pre><code>import os
from snakemake.io import expand

# This snakefile contains the processing steps of analysing 
# the 2x75 bp paired end of Lagomarsino et al. 2021. 
# 
# define some variables. 
pair_ids = [&quot;.r1&quot;, &quot;.r2&quot;]
ext = &quot;.fq.gz&quot;
FQC_EXT = [&quot;zip&quot;, &quot;html&quot;]

# some python code to extract the &quot;sample id&quot; from the file names. 
# Lachie wrote this for me since I&#39;m not a snake queen yet 
samples = os.listdir(&quot;01_rawdata/fastqraw&quot;)
samples = [sample.replace(ext, &quot;&quot;) for sample in samples]
for id in pair_ids:
    samples = [sample.replace(id, &quot;&quot;) for sample in samples]
samples = list(set(samples))

# here, we need to define the files which are the end products of the workflow. 
# Any files which are the input for a next step do not need to be included. 
# I have all of them in here, since I built this pipeline from scratch essentially. 
# This expand function will generate file names which alter based on these &quot;wildcardds&quot;. 
# A nice explanantion of how expand() works can be found here:
# https://endrebak.gitbooks.io/the-snakemake-book/content/chapters/expand/expand.html 

rule all:
    input:
        expand(&quot;01_rawdata/fastqc/{SAMPLE}{PAIR}_fastqc.{EXT}&quot;, SAMPLE = samples, PAIR = pair_ids, EXT = FQC_EXT),
        expand(&quot;02_trimdata/fastq/{SAMPLE}{PAIR_ID}{EXT}&quot;, SAMPLE = samples, PAIR_ID = pair_ids, EXT = ext),
        expand(&quot;02_trimdata/fastqc/{SAMPLE}{PAIR}_fastqc.{EXT}&quot;, SAMPLE = samples, PAIR = pair_ids, EXT = FQC_EXT),
        expand(&quot;03_alignstar/bam/{SAMPLE}.Aligned.sortedByCoord.out.bam&quot;, SAMPLE = samples),
        expand(&quot;03_alignstar/bam/{SAMPLE}.Aligned.sortedByCoord.out.bam.bai&quot;, SAMPLE = samples),
        expand(&quot;03_alignstar/FastQC/{SAMPLE}.Aligned.sortedByCoord.out_fastqc.{EXT}&quot;, SAMPLE = samples, EXT = FQC_EXT), 
        &quot;04_featureCounts/counts.out&quot;

# First, I will run fastqc on the raw data. 
rule fastqc:
    input:
        R1 = &quot;01_rawdata/fastqraw/{SAMPLE}.fq.gz&quot;
    params:
        outdir = &quot;01_rawdata/fastqc/&quot;
    output:    
        html = &quot;01_rawdata/fastqc/{SAMPLE}_fastqc.html&quot;,
        zip = &quot;01_rawdata/fastqc/{SAMPLE}_fastqc.zip&quot;
    conda:
        &quot;smk/envs/default.yaml&quot;
    resources: # parameters which will submit to phoenix
        cpu = 1,
        ntasks = 1,
        time = &quot;00-01:00:00&quot;,
        mem_mb = 4000
    shell:
        &quot;&quot;&quot;
        fastqc \
        -t {resources.cpu} \
        -o {params.outdir} \
        {input}
        &quot;&quot;&quot;

# The next step is to run adaptor removal using fastp. 
# I will only retain reads which are more than 20 nt in length after 
# trimming, and have a quality score of at least 15 phred. 
# I will also discard polyG reads, which I noticed are in this dataset. 

rule trim:
    input:
        R1 = &quot;01_rawdata/fastqraw/{SAMPLE}.r1.fq.gz&quot;,
        R2 = &quot;01_rawdata/fastqraw/{SAMPLE}.r2.fq.gz&quot;
    output:
        R1 = &quot;02_trimdata/fastq/{SAMPLE}.r1.fq.gz&quot;,
        R2 = &quot;02_trimdata/fastq/{SAMPLE}.r2.fq.gz&quot;,
        json = &quot;02_trimdata/log/{SAMPLE}.json&quot;,
        html = &quot;02_trimdata/log/{SAMPLE}.html&quot;
    params:
        bname = &quot;02_trimdata/fastq/{SAMPLE}&quot;
    conda:
        &quot;smk/envs/default.yaml&quot;
    resources:
        cpu = 1,
        ntasks = 1,
        time = &quot;00-01:00:00&quot;,
        mem_mb = 4000
    shell:
        &quot;&quot;&quot;        
        fastp \
            -l 20 \
            --json {output.json} \
            --html {output.html} \
            --detect_adapter_for_pe \
            --trim_poly_g \
            --out1 {output.R1} \
            --out2 {output.R2} \
            -i {input.R1} \
            -I {input.R2}
        &quot;&quot;&quot;
    # input:
    #     R1 = &quot;01_rawdata/fastqraw/{SAMPLE}.r1.fq.gz&quot;,
    #     R2 = &quot;01_rawdata/fastqraw/{SAMPLE}.r2.fq.gz&quot;
    # output:
    #     R1 = &quot;02_trimdata/fastq/{SAMPLE}.r1.fq.gz&quot;,
    #     R2 = &quot;02_trimdata/fastq/{SAMPLE}.r2.fq.gz&quot;
    # params:
    #     bname = &quot;02_trimdata/fastq/{SAMPLE}&quot;
    # conda:
    #     &quot;smk/envs/default.yaml&quot;
    # resources:
    #     cpu = 1,
    #     ntasks = 1,
    #     time = &quot;00-01:00:00&quot;,
    #     mem_mb = 4000
    # shell:
    #     &quot;&quot;&quot;        
    #     AdapterRemoval \
    #         --gzip \
    #         --trimns \
    #         --trimqualities \
    #         --minquality 30 \
    #         --minlength 20 \
    #         --basename {params.bname} \
    #         --threads {resources.cpu} \
    #         --output1 {output.R1} \
    #         --output2 {output.R2} \
    #         --file1 {input.R1} \
    #         --file2 {input.R2}
    #     &quot;&quot;&quot;
# repeat fastqc after trimming.         
rule trimqc:
    input:
        R1 = &quot;02_trimdata/fastq/{SAMPLE}.fq.gz&quot;
    params:
        outdir = &quot;02_trimdata/fastqc/&quot;  
    output:    
        html = &quot;02_trimdata/fastqc/{SAMPLE}_fastqc.html&quot;,
        zip = &quot;02_trimdata/fastqc/{SAMPLE}_fastqc.zip&quot;
    conda:
        &quot;smk/envs/default.yaml&quot;
    resources:
        cpu = 2,
        ntasks = 2,
        time = &quot;00-01:00:00&quot;,
        mem_mb = 6000
    shell:
        &quot;&quot;&quot;
        fastqc \
        -f fastq \
        -t {resources.cpu} \
        -o {params.outdir} \
        --noextract \
        {input}
        &quot;&quot;&quot;
        
rule align:
# here, we are aligning to the human genome (ensembll release 98). 
# The genome index is already generated in the /hpcfs folder by 
    input:
        R1 = &quot;02_trimdata/fastq/{SAMPLE}.r1.fq.gz&quot;, 
        R2 = &quot;02_trimdata/fastq/{SAMPLE}.r2.fq.gz&quot;       
    params:
        genomedir = &quot;/hpcfs/archive/biorefs/reference_genomes/ensembl-release-98/homo_sapiens/star/&quot;,
        bname = &quot;03_alignstar/bam/{SAMPLE}.&quot;
    output:    
        bam = &quot;03_alignstar/bam/{SAMPLE}.Aligned.sortedByCoord.out.bam&quot;
    conda:
        &quot;smk/envs/default.yaml&quot;
    resources:
        cpu = 16,
        ntasks = 1,
        time = &quot;00-05:00:00&quot;,
        mem_mb = 50000
    shell:
        &quot;&quot;&quot;
        STAR \
        --genomeDir {params.genomedir}\
        --runThreadN {resources.cpu} \
        --readFilesIn {input.R1} {input.R2} \
        --readFilesCommand &quot;gunzip -c&quot; \
        --outSAMtype BAM SortedByCoordinate \
        --outFileNamePrefix {params.bname}

        mkdir -p 03_alignstar/log
        mv {params.bname}*out 03_alignstar/log
        mv {params.bname}*tab 03_alignstar/log
        &quot;&quot;&quot;

rule fastqcalign:
    input:
        bam = &quot;03_alignstar/bam/{SAMPLE}.Aligned.sortedByCoord.out.bam&quot;
    output:
        &quot;03_alignstar/FastQC/{SAMPLE}.Aligned.sortedByCoord.out_fastqc.zip&quot;,
        &quot;03_alignstar/FastQC/{SAMPLE}.Aligned.sortedByCoord.out_fastqc.html&quot;
    params:
        outDir = &quot;03_alignstar/FastQC/&quot;
    conda:
        &quot;smk/envs/default.yaml&quot;
    resources:
        cpu = 2,
        ntasks = 2,
        time = &quot;00-01:00:00&quot;,
        mem_mb = 6000
    shell:
        &quot;&quot;&quot;
        fastqc -t {resources.cpu} \
        -o {params.outDir} \
        --noextract \
        {input.bam}
        &quot;&quot;&quot;

rule indexBam:
    input:
        &quot;03_alignstar/bam/{SAMPLE}.Aligned.sortedByCoord.out.bam&quot;
    output:
        &quot;03_alignstar/bam/{SAMPLE}.Aligned.sortedByCoord.out.bam.bai&quot;
    conda:
        &quot;smk/envs/default.yaml&quot;
    resources:
        cpu = 2,
        ntasks = 1, 
        time = &quot;00-01:00:00&quot;,
        mem_mb = 6000
    shell:
        &quot;&quot;&quot;
        samtools index {input} {output}
        &quot;&quot;&quot;


rule featureCounts:
    input:
        bam = expand(&quot;03_alignstar/bam/{SAMPLE}.Aligned.sortedByCoord.out.bam&quot;, SAMPLE = samples),
        gtf = &quot;Homo_sapiens.GRCh38.98.chr.gtf&quot; 
    output:
        counts = &quot;04_featureCounts/counts.out&quot;
    conda:
        &quot;smk/envs/default.yaml&quot;
    resources:
        cpu = 2,
        ntasks = 2,
        time = &quot;00-01:00:00&quot;,
        mem_mb = 6000
    params:
        minOverlap = 1,
        fracOverlap = 1,
        q = 10
# Settings for featureCounts. 
# To set this to count strictly exonic reads, I change fracOverlap to be the value 1. 
# The value minOverlap may also need adjusting based on your own read lengths. 
# the -Q option is set to 10, meaining a mapping quality of at least 10. 
# the -p flad indicates the input bam files were generated from paired end data
    shell:
       &quot;&quot;&quot;
       featureCounts \
       -Q {params.q} \
       -p \
       --minOverlap {params.minOverlap} \
       --fracOverlap {params.fracOverlap} \
       -T {resources.cpu} \
       -a {input.gtf} \
       -o {output} \
       {input.bam}
       &quot;&quot;&quot;</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.2 (2020-06-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Mojave 10.14.3

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.6.2

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.7        whisker_0.4       knitr_1.33        magrittr_2.0.1   
 [5] R6_2.5.0          rlang_0.4.11      fansi_0.5.0       stringr_1.4.0    
 [9] tools_4.0.2       xfun_0.24         utf8_1.2.1        git2r_0.28.0     
[13] jquerylib_0.1.4   htmltools_0.5.1.1 ellipsis_0.3.2    rprojroot_2.0.2  
[17] yaml_2.2.1        digest_0.6.27     tibble_3.1.2      lifecycle_1.0.0  
[21] crayon_1.4.1      later_1.2.0       sass_0.4.0        vctrs_0.3.8      
[25] promises_1.2.0.1  fs_1.5.0          glue_1.4.2        evaluate_0.14    
[29] rmarkdown_2.9     stringi_1.6.2     bslib_0.2.5.1     compiler_4.0.2   
[33] pillar_1.6.1      jsonlite_1.7.2    httpuv_1.6.1      pkgconfig_2.0.3  </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
