---
title: "initialQC"
author: "Karissa Barthelson"
date: "2021-10-22"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---
```{r setup, include=FALSE}
knitr::opts_chunk$set(
  autodep = TRUE,
  echo = TRUE,
  warning = FALSE,
  message = FALSE,
  fig.align = "center"
)
```

```{r loadLibs}
library(tidyverse)
library(magrittr)
library(pander)
library(AnnotationHub)
library(ggpubr)
library(ggfortify)
library(ggrepel)
library(ggeasy)
library(edgeR)
library(scales)
library(ngsReports)
library(pheatmap)

library(msigdbr)
```

```{r setOptions}
theme_set(theme_bw())
panderOptions("big.mark", ",")
panderOptions("table.split.table", Inf)
panderOptions("table.style", "rmarkdown")
```

## Introduction
```{r annotations}
# Note that this chunk takes a while to run, and requires internet connection
ah <- AnnotationHub() %>%
	subset(species == "Homo sapiens") %>%
	subset(rdataclass == "EnsDb")

ensDb <- ah[["AH75011"]] # for release 98, which is the genome version I aligned to
grTrans <- transcripts(ensDb)
trLengths <- exonsBy(ensDb, "tx") %>%
	width() %>%
	vapply(sum, integer(1))
mcols(grTrans)$length <- trLengths[names(grTrans)]
gcGene <- grTrans %>%
  mcols() %>%
  as.data.frame() %>%
  dplyr::select(gene_id, tx_id, gc_content, length) %>%
  as_tibble() %>%
  group_by(gene_id) %>%
  summarise(
    gc_content = sum(gc_content*length) / sum(length),
    length = ceiling(median(length))
  )
grGenes <- genes(ensDb)
mcols(grGenes) %<>%
  as.data.frame() %>%
  left_join(gcGene) %>%
  as.data.frame() %>%
  DataFrame()
```
  
In order to perform adequate QC, an `EnsDb` object was obtained for Ensembl release `r ensemblVersion(ensDb)` using the  `AnnotationHub` package.
This provided the GC content and length for each of the `r comma(length(grTrans))` transcripts contained in that release.


```{r dataImport}
meta <- 
  read_csv("data/confidentialData/ROSMAP-IN_assay_RNAseq_metadata.csv") %>% 
  dplyr::select(specimenID, RIN, rnaBatch, libraryBatch, sequencingBatch) %>% 
  left_join(read_csv("data/confidentialData/ROSMAP-IN_invitro_biospecimen_metadata.csv")) %>% 
  left_join(read_csv("data/confidentialData/ROSMAP-IN_individual_metadata.csv")) %>% 
  dplyr::select(where(~n_distinct(.) > 1)) %>% # omit the columns where its the same row
  left_join(read_csv("data/confidentialData/ROSMAP_clinical.csv"), by = "individualID") %>% 
  mutate(experimentor = str_extract(.$specimenID, 
                                    pattern = "(AH|VL)"), 
         temp = str_remove(.$specimenID, 
                           pattern = "(AH|VL)") %>% 
           str_remove(pattern = "_BR[0-9]+"), 
         sample = paste0(experimentor, ".s", temp), 
         apoeGenotype = case_when(
           apoeGenotype == 22 ~ "E2/E2", 
           apoeGenotype == 23 ~ "E2/E3", 
           apoeGenotype == 33 ~ "E3/E3", 
           apoeGenotype == 34 ~ "E3/E4",
           apoeGenotype == 44 ~ "E4/E4", 
         ),
         category = case_when( # from fig. 1B of Lag... et al. 
           specimenIdSource %in% c("BR01", "BR04", "BR08", "BR09", "BR13", "BR21", "BR24", "BR33", "BR40", "BR41", "BR58", "BR62", "BR65", "BR67", "BR68", "BR69", "BR72", "BR89", "BR93", "BR95") ~ "LP-NCI", 
           specimenIdSource %in% paste0("BR", c(11, 22, 26, 28, 29, 30, 36, 37, 43, 48, 50, 59, 60, 61, 63, 64, 91)) ~ "HP-NCI", # a better way to do this.
           specimenIdSource %in% paste0("BR", c(14, 15, 27, 39, 46, 54, 57, 66, 83, 92, 97:99, 103:105, 108)) ~ "AD"
         
         )
  ) %>% 
   dplyr::select(where(~n_distinct(.) > 1)) %>% # omit the columns where its the same row
  dplyr::select(sample, experimentor, everything())# make some new cols to match the names in the gene counts matrix


featureCounts <- 
  read_delim("data/confidentialData/counts.out", delim = "\t", skip = 1) %>%
  set_names(basename(names(.))) %>% 
  as.data.frame() %>%
  dplyr::select(-c(Chr, Start, End, Length, Strand)) %>% 
  as_tibble %>% 
  column_to_rownames("Geneid")

# tidy up colnames
colnames(featureCounts) %<>% 
  str_remove(".Aligned.sortedByCoord.out.bam")
```

# raw data QC

```{r}
fastqc_raw <- list.files(
    path = "data/confidentialData/fastqc_raw/",
    pattern = "zip", 
    recursive = TRUE,
    full.names = TRUE
    ) %>% 
    FastqcDataList()
```

## GC content
Strange peaks are observed with 100% GC content. After a quick google search, it appears these are sequeniccng artefacts. 

```{r}
plotGcContent(
  x = fastqc_raw, 
  plotType = "line",
  gcType = "Transcriptome"
) +
  theme(legend.position = "none")
```

## Over-represented sequences 

There doesn't appear to be any over-represented sequences which are worrying. The top ones are mostly adaptors which are removed in the `snakemake` pipeline. 

```{r}
getModule(fastqc_raw, "Overrep") %>% 
    group_by(Sequence, Possible_Source) %>% 
    summarise(`Found In` = n(), `Highest Percentage` = max(Percentage)) %>% 
    arrange(desc(`Highest Percentage`), desc(`Found In`)) %>% 
    ungroup() %>% 
    dplyr::slice(1:30) %>%
    mutate(`Highest Percentage` = percent_format(0.01)(`Highest Percentage`/100)) %>%
    pander(
        justify = "llrr",
        caption = paste(
            "*Top", nrow(.),"Overrepresented sequences.",
            "The number of samples they were found in is shown,",
            "along with the percentage of the most 'contaminated' sample.*"
            )
    )
```

# Trimmed data QC

```{r trimStats}
fastqc_trim <- list.files(path = "data/confidentialData/fastqc_trim", 
                          pattern = "zip", 
                          full.names = TRUE) %>%
  FastqcDataList()

trimStats <- readTotals(fastqc_raw) %>%
  dplyr::rename(Raw = Total_Sequences) %>%
  left_join(readTotals(fastqc_trim), by = "Filename") %>%
  dplyr::rename(Trimmed = Total_Sequences) %>%
  dplyr::filter(grepl("r1", Filename)) %>%
  mutate(
    Discarded = 1 - Trimmed / Raw,
    Retained = Trimmed / Raw
  )
```

The raw reads were processed with [fastp](https://github.com/OpenGene/fastphttps://github.com/OpenGene/fastp). Reads which contained an average quality score of < 15 were omitted, as were reads which were shorter than 20nt after adaptor removal. I allowed fastp to detect adaptors automatically. After adapter trimming between `r pander(range(percent_format(0.01)(trimStats$Discarded)))` of reads were discarded. No over-represented sequences remained, and the %GC appears to have improved. 

```{r}
ggarrange(
  plotGcContent(
    x = fastqc_raw, 
    plotType = "line",
    gcType = "Transcriptome"
  ) +
    theme(legend.position = "none") +
    ggtitle("Before trimming/filtering"), 
  plotGcContent(
  x = fastqc_trim, 
  plotType = "line",
  gcType = "Transcriptome"
) +
  theme(legend.position = "none")+
  ggtitle("After trimming/filtering")
) 
```

# Align QC

Trimmed reads were aligned to the human genome (Ensembl relealease 98) using `STAR 2.7.0d` and summarised to each gene using `featureCounts`. These counts were to be used for all gene-level analysis. 
```{r}
fastqc_align <- list.files(path = "data/confidentialData/fastqc_align", 
                          pattern = "zip", 
                          full.names = TRUE) %>%
  FastqcDataList()
```

```{r}
plotGcContent(
  x = fastqc_align, 
  plotType = "line",
  gcType = "Transcriptome"
) +
  theme(legend.position = "none")
```

# Filtering lowly expressed genes

Genes which are lowly expressed are uninformative for DE analysis. Here, I will consider a gene to be lowly expressed if it contains a CPM of less than 2 in at least 33 samples (i.e. the number of AD samples). The effect of filtering is be shown in the density plots below. 

```{r}
a <- featureCounts %>% 
  cpm(log = TRUE) %>%
  as.data.frame() %>% 
  pivot_longer(
    cols = everything(),
    names_to = "sample",
    values_to = "logCPM"
  ) %>%
  split(f = .$sample) %>%
  lapply(function(x){
    d <- density(x$logCPM)
    tibble(
      sample = unique(x$sample),
      x = d$x,
      y = d$y
    )
  }) %>%
  bind_rows() %>%
  left_join(meta) %>% 
  ggplot(aes(x, y, colour = diagnosis, group = sample)) +
  geom_line() +
  labs(
    x = "logCPM",
    y = "Density",
    colour = "category"
  )+
  ggtitle("Before filtering")

b <- featureCounts %>% 
  .[rowSums(cpm(.) >= 2) >= 33,] %>% 
  cpm(log = TRUE) %>%
  as.data.frame() %>% 
  pivot_longer(
    cols = everything(),
    names_to = "sample",
    values_to = "logCPM"
  ) %>%
  split(f = .$sample) %>%
  lapply(function(x){
    d <- density(x$logCPM)
    tibble(
      sample = unique(x$sample),
      x = d$x,
      y = d$y
    )
  }) %>%
  bind_rows() %>%
  left_join(meta) %>% 
  ggplot(aes(x, y, colour = diagnosis, group = sample)) +
  geom_line() +
  labs(
    x = "logCPM",
    y = "Density",
    colour = "category"
  )+
  ggtitle("After filtering") 
ggarrange(a, b, common.legend = TRUE)
```

```{r dge}
dge <- featureCounts %>% 
  as.matrix() %>% 
  .[rowSums(cpm(.) >= 2) >= 33,] %>% 
  DGEList(
    samples = tibble(sample = colnames(.)) %>%
      left_join(meta) %>% 
      as_tibble(),
    genes = grGenes[rownames(.)] %>%
      as.data.frame() %>%
      dplyr::select(
        chromosome = seqnames, start, end, 
        gene_id, gene_name, gene_biotype, description, entrezid
      ) %>% 
      left_join(gcGene) %>% 
      as_tibble()
  ) %>%
  calcNormFactors()
```

# Check library sizes

Libary sizes vary considerably, which appears to be due to samples being sequenced across multiple runs. Library sizes range between `r pander(comma(range(dge$samples$lib.size)))`. Although concerning, this should be accounted for using `TMM normalisation`.
```{r}
dge$samples %>% 
  ggplot(aes(x = specimenIdSource, y = lib.size, fill = sequencingBatch)) + 
  geom_col() +
  easy_rotate_x_labels(angle = -45) +
  facet_wrap(~category, scales = "free_x") +
  scale_y_continuous(labels = comma) +
  labs(
    x = "Sample", 
    y = "RNA-seq library size", 
    fill = "RNA-seq batch"
  )
```

# PCA
```{r dgeFilt}
dge <- featureCounts %>% 
  as.matrix() %>% 
  .[rowSums(cpm(.) >= 2) >= 33,] %>% 
  DGEList(
    samples = tibble(sample = colnames(.)) %>%
      left_join(meta) %>% 
      as_tibble(),
    genes = grGenes[rownames(.)] %>%
      as.data.frame() %>%
      dplyr::select(
        chromosome = seqnames, start, end, 
        gene_id, gene_name, gene_biotype, description, entrezid
      ) %>% 
      left_join(gcGene) %>% 
      as_tibble()
  ) %>%
  calcNormFactors()
```

I next performed a PCA on the logCPM values after filtering lowly expressed genes. I want to see how similar the samples look. Two distinct clusters are observed across PC2 (explaining ~19% of the total variation in the dataset). 

```{r}
cpm(dge, log = TRUE) %>%
  t() %>%
  prcomp() %>% 
  autoplot(data = tibble(sample = rownames(.$x)) %>%
             left_join(dge$samples),
           colour = "experimentor", 
           size = 4,
           shape = "sex")
```

I next want to see how similar the technical replicates are. In the plot below, only samples which were sequenced twice are labelled. Most technical replicates do not cluster together, meaning they are not overly similar. 

```{r}
cpm(dge, log = TRUE) %>%
  t() %>%
  prcomp() %>% 
  autoplot(data = tibble(sample = rownames(.$x)) %>%
             left_join(dge$samples),
             size = 2,
           colour = "specimenIdSource",
             shape = "sex") +
    geom_label_repel(aes(label = specimenIdSource, 
                         colour = specimenIdSource), 
                     max.overlaps = 15,
                     data = . %>% 
                       dplyr::filter(specimenIdSource %in% dge$samples$specimenIdSource[dge$samples %>% dplyr::select("specimenIdSource") %>% duplicated()])
                     )
```

# Why do the technical replicates look so different?

One would expect that technical replicates would look mostly similar since they were derived from the same sample of iPSC neurons. But the PCA plot above shows that this is not the case. From inspection of the metadata about the samples, looks like the RNA integrity number (RIN) is fairly low in one of the library preparation batches. 

```{r}
techRep_brID <- dge$samples$specimenIdSource[dge$samples %>% dplyr::select("specimenIdSource") %>% duplicated()]

techRep_sampleID <- dge$samples %>% 
  dplyr::filter(specimenIdSource %in% techRep_brID) %>% 
  .$sample
 
dge$samples %>%  
  dplyr::filter(specimenIdSource %in% techRep_brID) %>% 
  ggplot(aes(x = sample, y = lib.size/1e6, fill = RIN)) +
  geom_col() +
  facet_wrap(~libraryBatch, scales = "free_x") +
  scale_fill_viridis_c() +
  labs(y = "Library size (millions)") +
  easy_rotate_x_labels(angle = -45) +
  ggtitle("Library sizes of samples which had technical replicates,\ncoloured by RNA integrity", 
          subtitle = "Samples are grouped by library batch")
```

Inspection of the two replicates for each iPSC neuron samples shows that most of the time (except for BR14, which had fairly similar RINs), there is onw "high quality" library, and one with lesser quality. 

```{r}
dge$samples %>%  
  dplyr::filter(specimenIdSource %in% techRep_brID) %>% 
  ggplot(aes(x = sample, y = lib.size/1e6, fill = RIN)) +
  geom_col() +
  facet_wrap(~specimenIdSource, scales = "free_x") +
  scale_fill_viridis_c() +
  labs(y = "Library size (millions)") +
  easy_rotate_x_labels(angle = -45)
```

```{r}
samps2keep <- dge$samples %>% 
  group_by(specimenIdSource) %>% 
  dplyr::slice(which.max(RIN)) %>% 
  .$sample
dge <- dge[,samps2keep]

```

Therefore, for this analysis, I will omit the technical replicate which had the lower RIN for each iPSC sample. This leaves `r dim(dge)[2]` samples out of the original `r dim(meta)[1]`. This is a different strategy to [Lagomarsino et al. 2021](https://doi.org/10.1016/j.neuron.2021.08.003), who averaged out their replicates. I dont think this is the best way to deal with this, as the RNA was degraded in one of the technical replicates. This may influence how the reads are generated during the sequencing, and avergaing it out across a better quality sample may bring about false results.  

# Assessment of differentiated cell lines. 

[Lagomarsino et al. 2021](https://doi.org/10.1016/j.neuron.2021.08.003) noted in their methods section that a few samples may show abnormally high levels of Oct2 (SLC22A2), Oct3 (SLC22A3) or LEFTY2. This implies that these cell lines may contain a high proportion of undifferentiated cells and should be removed. These three genes were filtered from the analysis in the filtering step, meaning they are lowly expressed and considered undetectable. Notably, the OCT2/3 genes show fairly consistent expression across the samples. 

The samples show pretty variable expression of LEFTY2. Basing a decision on whether to keep or omit a sample based on the expression of 1 gene seems a bit dodgy. So I will also look at three other common pluripotnecy genes: NANOG, OCT4 and ZFP42 (REX1). 

The graphs below show the expression of these genes from the unfiltered expression levels.
 
```{r}
# Obtain the gene ids for the marker genes
undifferentiatedMarkerGeneIDs <- 
  grGenes %>% 
  as_tibble() %>% 
  dplyr::filter(gene_name %in% c("LEFTY2", "SLC22A2", "SLC22A3", "NANOG", "POU5F1", "ZFP42")) %>% 
  na.omit() %>% 
  .$gene_id

# PLot the expression *before filtering
featureCounts %>% 
  cpm(log=T) %>% 
  as.data.frame() %>% 
  .[undifferentiatedMarkerGeneIDs,samps2keep] %>% 
  rownames_to_column("gene_id") %>% 
  gather(key = "sample", value = "logCPM", contains("s")) %>% 
  left_join(grGenes %>% as_tibble() %>% dplyr::select(gene_id, gene_name)) %>% 
  as_tibble() %>% 
  na.omit() %>% 
  left_join(meta) %>% 
  ggplot(aes(x = sample, y = logCPM)) +
  geom_col() +
  geom_col(aes(fill = specimenIdSource), 
           data = . %>%  dplyr::filter(logCPM > 1)
  ) + 
   geom_label(aes(label = specimenIdSource, colour = specimenIdSource), 
             data = . %>%  dplyr::filter(logCPM > 1), 
             position = position_nudge(y = 0.5), 
             show.legend = FALSE) +
  facet_wrap(~gene_name) +
  easy_rotate_x_labels(angle = -45) + 
  labs(fill = "Samples with potentiallt high expression of undifferentiated marker genes")+
  ggtitle("logCPM (before filtering) of pluripotency marker genes") +
  theme(legend.position = "none")



cpm(dge, log = TRUE) %>%
  t() %>%
  prcomp() %>% 
  autoplot(data = tibble(sample = rownames(.$x)) %>%
             left_join(dge$samples),
           colour = "libraryBatch" , 
           size = 4,
           shape = "category") +
  geom_label_repel(aes(label = specimenIdSource), 
             data = . %>%  dplyr::filter(specimenIdSource %in% c("BR62", "BR65")), 
             nudge_y = -0.1, 
             nudge_x =  -0.1,
             show.legend = FALSE) 
```

An alternative way to assess the purity of these samples is to assess marker genes of neurons. Here, I obtained the gene set from Cahoy et al from msigdb and will assess the expression of these genes in the iPSC neuron samples. 

```{r}
cahoy_neurons <- msigdbr("Homo sapiens", category = "C6") %>% 
  dplyr::filter(grepl(gs_name, pattern = "CAHOY_NEURONAL")) %>% 
  dplyr::distinct(gs_name, ensembl_gene, .keep_all = TRUE) %>% 
  .$ensembl_gene

dge %>% 
  cpm(log=T) %>% 
  as.data.frame() %>% 
  .[cahoy_neurons,] %>% 
  na.omit %>% # Omit the genes undetectable in the experinment
  t %>% 
  prcomp %>% 
  autoplot(data = tibble(sample = rownames(.$x)) %>%
             left_join(dge$samples),
           colour = "category", 
           shape = "sex",
           size = 4) + 
  geom_label_repel(aes(label = specimenIdSource), 
                   data = . %>%  dplyr::filter(specimenIdSource %in% c("BR62", "BR65")), 
                   nudge_y = -0.1, 
                   nudge_x =  -0.1,
                   show.legend = FALSE) +
  ggtitle("PCA of 75 neuronally expressed genes in iPSC cells")
          
```

I dont think there is enough evidence here to determine whether any particular sample contains too many undifferentiated iPSC cells to omit it. 

# PCA after omitting samples
The PCA was repeated after omitting the problematic RNA-seq libraries (technical replicates). Samples still seperate by library batch. So I will need to correct for this

```{r PCAafterOmit}
cpm(dge, log = TRUE) %>%
  t() %>%
  prcomp() %>% 
  autoplot(data = tibble(sample = rownames(.$x)) %>%
             left_join(dge$samples),
           colour = "category", 
           size = 4) +
  scale_colour_viridis_d(end = 0.8) +
  theme(aspect.ratio = 1) +
  ggtitle("AD?")

cpm(dge, log = TRUE) %>%
  t() %>%
  prcomp() %>% 
  autoplot(data = tibble(sample = rownames(.$x)) %>%
             left_join(dge$samples),
           colour = "sex", 
           size = 4) +
  scale_colour_viridis_d(end = 0.8) +
  theme(aspect.ratio = 1) +
  ggtitle("Sex")
  
cpm(dge, log = TRUE) %>%
  t() %>%
  prcomp() %>% 
  autoplot(data = tibble(sample = rownames(.$x)) %>%
             left_join(dge$samples),
           colour = "libraryBatch",                
           
           size = 4) +
  scale_colour_viridis_d(end = 0.9) +
  theme(aspect.ratio = 1) +
  ggtitle("RNA-seq library batch")

cpm(dge, log = TRUE) %>%
  t() %>%
  prcomp() %>% 
  autoplot(data = tibble(sample = rownames(.$x)) %>%
             left_join(dge$samples),
           colour = "RIN", 
           size = 4) +
  scale_color_viridis_c() +
  theme(aspect.ratio = 1) +
  ggtitle("RNA Integrity Number (RIN)")

cpm(dge, log = TRUE) %>%
  t() %>%
  prcomp() %>% 
  autoplot(data = tibble(sample = rownames(.$x)) %>%
             left_join(dge$samples),
           colour = "Study", 
           size = 4) +
  scale_color_viridis_d(end = 0.9) +
  theme(aspect.ratio = 1) +
  ggtitle("ROS or MAP cohort")

cpm(dge, log = TRUE) %>%
  t() %>%
  prcomp() %>% 
  autoplot(data = tibble(sample = rownames(.$x)) %>%
             left_join(dge$samples),
           colour = "age_death", 
           size = 4) +
  scale_color_viridis_d(end = 0.9) +
  theme(aspect.ratio = 1) +
  ggtitle("Age at death")
```

```{r}
cpm(dge, log = TRUE) %>%
  t() %>%
  prcomp() %>% 
  .$x %>% 
  as.data.frame() %>% 
  rownames_to_column("sample") %>% 
  left_join(dge$samples) %>% 
  mutate(`Library Size (millions) ` = lib.size/1e6) %>% 
  ggplot(aes(x = PC1, y = `Library Size (millions) `)) +
  geom_point(aes( colour = category, shape = sex), 
             size = 4) +
  geom_smooth(method = "lm") +
  theme_bw()

cpm(dge, log = TRUE) %>%
  t() %>%
  prcomp() %>% 
  .$x %>% 
  as.data.frame() %>% 
  rownames_to_column("sample") %>% 
  left_join(dge$samples) %>% 
  mutate(`Library Size (millions) ` = lib.size/1e6) %>% 
  ggplot(aes(x = PC2, y = `Library Size (millions) `)) +
  geom_point(aes(colour = diagnosis, shape = sex), 
             size = 4) +
  geom_smooth(method = "lm") +
  theme_bw()

cpm(dge, log = TRUE) %>%
  t() %>%
  prcomp() %>% 
  .$x %>% 
  as.data.frame() %>% 
  rownames_to_column("sample") %>% 
  left_join(dge$samples) %>% 
  mutate(`Library Size (millions) ` = lib.size/1e6) %>% 
  ggplot(aes(x = PC1, y = RIN)) +
  geom_point(aes( colour = diagnosis, shape = sex), 
             size = 6) +
  geom_smooth(method = "lm") +
  theme_bw()

cpm(dge, log = TRUE) %>%
  t() %>%
  prcomp() %>% 
  .$x %>% 
  as.data.frame() %>% 
  rownames_to_column("sample") %>% 
  left_join(dge$samples) %>% 
  mutate(`Library Size (millions) ` = lib.size/1e6) %>% 
  ggplot(aes(x = PC2, y = RIN)) +
  geom_point(aes( colour = diagnosis, shape = sex), 
             size = 6) +
  geom_smooth(method = "lm") +
  theme_bw() 
```

Might be a trend happening between the second principal component and the RIN of the sample. 

# Conclusion

I next need to remove batch effects. This will be explored in the [Batch effects analysis](removeBatchEffects.html).   


```{r exportData}
saveRDS(dge, "data/confidentialData/dge.rds")
```


