---
title: "Home"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: false
editor_options:
  chunk_output_type: console
---
# IRE analysis of LOAD patient-derived iPSC neurons 
In this analysis, I will re-analyse the dataset generated by [Lagomarsino et al. 2021](https://doi.org/10.1016/j.neuron.2021.08.003). They used cryopreserved peripheral blood mononuclear cells donated by 53 participants in the Religious Orders Study and Rush Memory and Aging Project (ROSMAP) to generate iPSC lines. They then generated neuronal cultures from these cells that were 98% pure, with gene-expression profiles similar to those of cortical glutamatergic neurons. These neuronal cultures were then subjected to bulk RNA-seq. Raw data is available from [synapse](https://www.synapse.org/#!Synapse:syn23564891). Note that permission is required to access the raw data.

Here, I will re-analyse this dataset to determine whether we can detect changes to the expression of genes which contain iron-reponsive elements (IREs) in their transcripts. I will use the gene sets developed by [Hin et al. 2020](https://www.biorxiv.org/content/10.1101/2020.05.01.071498v5), and examine there expression across the LOAD iPSC neurons. 


